services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3622,10 +3622,16 @@ private boolean whetherToCreateNewDataRecoveryVersion(
       String clusterName,
       StoreInfo destStore,
       int versionNumber) {
-    // Currently new version data recovery is only supported for batch-only store.
-    // For existing data centers, current store version might be serving read requests. Need to create a new version.
-    // For new data centers or non-current version, it's ok to delete and recreate it. No need to create a new version.
-    return destStore.getHybridStoreConfig() == null && versionNumber == destStore.getCurrentVersion()
+    /**
+     * Creating a new data recovery version on the destination colo when satisfying:
+     * 1. New version data recovery is only supported for batch-only store.
+     * 2. For the existing destination data center, a new version is needed if
+     *    2.1. srcVersionNumber equals to the current version in dest colo, as current version is serving read requests.
+     *    2.2. srcVersionNumber is less than the current version in dest colo, because Venice normally assumes that a
```

Comment:
Just to make sure that I understand the question accurately, let me give a concrete example for the ease of discussion, assuming src current ver = 19, dest current ver = 21 (but has data loss issue), and we want to recover the dest from data loss by using the src. What would happen after this change is that we create a new version in the dest, so dest cur version = 22. 

So, with this example, is the question about why not use ver = 21 in the dest for recovery (e.g. delete it and then recreate it)? To that, I think there was concern about deleting the current version in dest because it may still serve read requests.